htdocs/fourn/facture/card.php

```diff
@@ -527,14 +527,13 @@
 			$i = 0;
 			foreach ($object->lines as $line) {
 				if ($line->product_type < 9 && $line->total_ht != 0) { // Remove lines with product_type greater than or equal to 9 and no need to create discount if amount is null
-					$keyforvatrate = $line->tva_tx.($line->vat_src_code ? ' ('.$line->vat_src_code.')' : '');
 
 					$amount_ht[$line->tva_tx] += $line->total_ht;
 					$amount_tva[$line->tva_tx] += $line->total_tva;
 					$amount_ttc[$line->tva_tx] += $line->total_ttc;
-					$multicurrency_amount_ht[$keyforvatrate] += $line->multicurrency_total_ht;
-					$multicurrency_amount_tva[$keyforvatrate] += $line->multicurrency_total_tva;
-					$multicurrency_amount_ttc[$keyforvatrate] += $line->multicurrency_total_ttc;
+					$multicurrency_amount_ht[$line->tva_tx] += $line->multicurrency_total_ht;
```

Comment:
If i compare with the customer side, it seems the fix must be done into the opisite direction:
Replacing  $line->tva_tx  with  $keyforvatrate  into  $amount_xxx[]  instead of changing the multicurrency_amount_ht

Reply:
@eldy changes have been done according to your recommendations